test/functional/output_forward.go

```diff
@@ -27,10 +34,31 @@ const unsecureFluentConf = `
 		@type json
 	</format>
 </match>
-<match **>
-	@type stdout
+
+<filter linux-audit.log**>
```

Comment:
Why are we doing any special parsing/processing the message in the output container?  One point of doing functional tests is to verify we are generating our normalization and collection pipeline correctly.  Needing to parse the output shipped from the main fluent container would leave me to believe:
* We are not normalizing something correctly
* We are unnecessarily post-processing something for the test

Reply:
@jcantrill apparently the forwarded time_key format as-is for "linux-audit.log" do not work with @type file to write logs to file ... so you **can't** write the output format of  "linux-audit.log" to file ... so ... I spend almost two working days trying to understand what is going on here. I learned that you are using proprietary RH parser to parse the logs and do not use the required "%Y-%m-%dT%H:%M:%S.%N%z" (somewhere in the ViaQ github project) and to overcome that I had to change the format reading time from @timestamp and writing to time_key is the correct format ...  I did not want (or know if this make sense) to change the parsing code ...

Reply:
@jcantrill @alanconway  I invested more time today to make this PR even more robust and make the tests easier to create ... look on the templates ... they are very easy to create right now .... 

The big challenge for me is that I do not know what are the correct formats ... so with this PR we have all the building blocks ... but now we need to feed this will some more complicated formats to check ... and make sure that all output fields are as expected. I can just see what is the output and compare but I am not sure that this is correct output

Reply:
@alanconway look also on prev. comment

Reply:
We really need to review & lay out the data model properly, it's on my TODO list. 

Meantime how about we reduce the expectations of the test to just items we are confident about, and extend it later with more details.

My concern with the template appproach is that it works for equality/regexp tests on string values, but it's no use for complicated non-string values like time.Time for instance. However I think it's worth merging this and improving later, since the test already provides some value.

/lgtm

Reply:
+1, I say again, lets merge it and improve it as we go.

Reply:
/lgtm